enterprise/coderd/roles.go

```diff
@@ -131,6 +132,85 @@ func (api *API) patchOrgRoles(rw http.ResponseWriter, r *http.Request) {
 	httpapi.Write(ctx, rw, http.StatusOK, db2sdk.Role(inserted))
 }
 
+// deleteOrgRole will remove a custom role from an organization
+//
+// @Summary Delete a custom organization role
+// @ID delete-a-custom-organization-role
+// @Security CoderSessionToken
+// @Produce json
+// @Param organization path string true "Organization ID" format(uuid)
+// @Param roleName path string true "Role name"
+// @Tags Members
+// @Success 200 {array} codersdk.Role
+// @Router /organizations/{organization}/members/roles/{roleName} [delete]
+func (api *API) deleteOrgRole(rw http.ResponseWriter, r *http.Request) {
+	var (
+		ctx               = r.Context()
+		auditor           = api.AGPL.Auditor.Load()
+		organization      = httpmw.OrganizationParam(r)
+		aReq, commitAudit = audit.InitRequest[database.CustomRole](rw, &audit.RequestParams{
+			Audit:          *auditor,
+			Log:            api.Logger,
+			Request:        r,
+			Action:         database.AuditActionDelete,
+			OrganizationID: organization.ID,
+		})
+	)
+	defer commitAudit()
+
+	rolename := chi.URLParam(r, "roleName")
+	roles, err := api.Database.CustomRoles(ctx, database.CustomRolesParams{
+		LookupRoles: []database.NameOrganizationPair{
+			{
+				Name:           rolename,
+				OrganizationID: organization.ID,
+			},
+		},
+		ExcludeOrgRoles: false,
+		OrganizationID:  organization.ID,
```

Comment:
weird duplication of `OrganizationID` 🤔

Reply:
Yea, it actually is not necessary, but we have a linter that makes you fill out all fields in a struct. This felt more intuitive that doing `uuid.Nil`. But you have to do something, or the linter yells at you :cry:

Reply:
tldr on why this field can't be removed?